src/artifact.rs

```diff
@@ -262,8 +266,9 @@ impl Artifact {
         match self.declarations.get(&decl_name) {
             Some(ref stype) => {
                 let prop = match *stype {
-                    &Decl::Data { global } => Prop { global, function: false },
-                    &Decl::Function { global } => Prop { global, function: true },
+                    &Decl::CString { global } => Prop { global, function: false, writeable: false, cstring: true },
+                    &Decl::Data { global, writeable } => Prop { global, function: false, writeable, cstring: false },
+                    &Decl::Function { global } => Prop { global, function: true, writeable: false, cstring: false},
```

Comment:
I think it may soon be time to refactor these flags into variants but will have to put some thought into it, and not be onerous to the clients.

Then again prop should only be seen by backends so it’s not as bad. Dunno must think